internal/rbd/rbd_attach.go

```diff
@@ -259,6 +263,10 @@ func appendNbdDeviceTypeAndOptions(cmdArgs []string, isThick bool, userOptions s
 			cmdArgs = append(cmdArgs, "--options", fmt.Sprintf("%s=%d", setNbdIOTimeout, defaultNbdIOTimeout))
 		}
 
+		if cookie != "" && hasNBDCookieSupport {
```

Comment:
instead of repeating this if statement, add a helper:
```go
cmdArgs = appendRbdNbdCookie(cmdArgs, cookie)

func appendRbdNbdCookie(...) []string {
    if cookie != "" && hasNBDCookieSupport {
        cmdArgs = ....
    }
    return cmdArgs
}
```

Reply:
I don't see much value @nixpanic 

The syntax of cmdArgs in both cases are different

one is:

```
if cookie != "" && hasNBDCookieSupport {
        cmdArgs = append(cmdArgs, "--options", fmt.Sprintf("cookie=%s", cookie))
}

```

and the other is:
```
if cookie != "" && hasNBDCookieSupport {
        cmdArgs = append(cmdArgs, fmt.Sprintf("--cookie=%s", cookie))
}
```

Reply:
We need to defent further with a bool again, I would prefer keep it as it is just like others in the respective functions.

Reply:
@nixpanic actually we don't have to check for empty `cookie` as it is passed from VolID which will never be empty?